sql/src/main/antlr/OpenSearchSQLParser.g4

```diff
@@ -222,10 +222,10 @@ nullLiteral
 
 // Date and Time Literal, follow ANSI 92
 datetimeLiteral
-    : dateLiteral
-    | timeLiteral
-    | timestampLiteral
-    | datetimeConstantLiteral
+    : dateLiteral #pasthrough
+    | timeLiteral #passthrough
+    | timestampLiteral #passthrough
+    | datetimeConstantLiteral #functionShortcut
```

Comment:
Why?

Reply:
`#passthrough` is needed to satisfy ANTLR -- if one alternative is labeled, all of them have to be, but the labels don't have to be unique. I'll add a comment.

`#functionShortcut` label is used to implement function shortcuts -- `NOW` to mean `NOW()`. If this alternative is taken, `visitFunctionShortcut` will be called which will create a `Function` node with value of `datetimeConstantLiteral` as function name.
https://github.com/Bit-Quill/opensearch-project-sql/blob/c0c3188e7946113d1140d059a1ee491748f15972/ppl/src/main/java/org/opensearch/sql/ppl/parser/AstExpressionBuilder.java#L338-L341
and
https://github.com/Bit-Quill/opensearch-project-sql/blob/c0c3188e7946113d1140d059a1ee491748f15972/ppl/src/main/java/org/opensearch/sql/ppl/parser/AstExpressionBuilder.java#L257-L266

Reply:
... of course, now I realize this would've been better implemented as `ReferenceExpression`.  Ah well -- room for improvement.

Reply:
I was surprised, because #92 worked without labels. There were another mechanism involved. Thank you for explanation!

Reply:
Hm, I thought this was handled by value-caching for constant functions. I'll double check.